### PR TITLE
refactor system/integration auth flow to use OAuth 2.0

### PR DESCRIPTION
**Pull request by naftis**
#5830

Refactors the auth microservice to return OAuth 2.0 compliant token responses. I've followed the documentation here: https://www.oauth.com/oauth2-servers/access-tokens/access-token-response/

I also included a refactor commit which moves advanced search helpers away from FHIR files as they shouldn't be related.

### CODE REVIEW

File: packages/auth/src/features/system/handler.test.ts [hunk 1]
```diff
@@ -26,11 +26,7 @@ describe('authenticate handler receives a request', () => {
       fetch.mockReject(new Error())
       const res = await server.server.inject({
         method: 'POST',
-        url: '/authenticateSystemClient',
-        payload: {
-          client_id: '123',
-          client_secret: '456'
-        }
+        url: '/token?client_id=123&client_secret=456&grant_type=client_credentials'
```
Comment (Zangetsu101) [inline packages/auth/src/features/system/handler.test.ts:29]: Is it required for the client_id & client_secret to be passed as search params? It's not even a GET request

Comment (Zangetsu101) [reply packages/auth/src/features/system/handler.test.ts:29]: hm I guess that's the standard

**Review by Zangetsu101 — APPROVED** [review]
Looks good @naftis. Could you please update the postman collections on Farajaland to use this new flow?

File: packages/auth/src/features/system/handler.test.ts (hunk 1 above)
Comment (naftis) [reply packages/auth/src/features/system/handler.test.ts:29]: Yeah, assuming that's the standard as it's exampled [here](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/)

**Pull request merged**